spec/models/phone_number_spec.rb

```diff
@@ -4,55 +4,132 @@
   describe 'validations' do
-    let(:user) do
-      User.create(username: 'Mike',
-                  biography: 'I am a developer',
-                  age: 25)
+    subject do
+      PhoneNumber.new(phone_number: '123',
+                      label: 0,
+                      main: true,
+                      contact_id: contact.id)
     end
 
-    let(:contact) do
-      Contact.create(name: 'Rudolph',
-                     user_id: user.id)
+    let(:user) do
+      User.create!(username: 'Mike',
+                   biography: 'I am a developer',
+                   age: 25)
     end
 
-    subject do
-      PhoneNumber.new(phone_number: '00123456789',
-                      label: 'work',
-                      primary: FalseClass,
-                      contact_id: contact.id)
+    let(:contact) do
+      Contact.create!(name: 'Rudolph',
+                      user_id: user.id)
     end
 
-    context 'when all attributes are present' do
+    context 'when needed attributes are present' do
       it 'is valid' do
         expect(subject.valid?).to be_truthy
       end
     end
 
     context 'when number is not present' do
+      before { subject.phone_number = nil }
+
       it 'is not valid without a phone_number' do
-        subject.phone_number = nil
         expect(subject.valid?).to be_falsey
+        expect(subject.errors.full_messages).to include("Phone number can't be blank")
+      end
+    end
+
+    context 'when phone_number has already been taken' do
+      before do
+        PhoneNumber.create!(phone_number: '123',
+                            label: 0,
+                            contact_id: contact.id)
+      end
+
+      it 'is not valid to use a taken phone_number' do
+        expect(subject.valid?).to be_falsey
+        expect(subject.errors.full_messages).to include('Phone number has already been taken')
       end
     end
 
     context 'when label is not present' do
+      before { subject.label = nil }
+
       it 'is not valid without a label' do
-        subject.label = nil
         expect(subject.valid?).to be_falsey
+        expect(subject.errors.full_messages).to include("Label can't be blank")
       end
     end
 
-    context 'when primary is not present' do
+    context 'when main is not present' do
+      before { subject.main = nil }
+
       it 'is not valid without primary' do
-        subject.primary = nil
         expect(subject.valid?).to be_falsey
+        expect(subject.errors.full_messages).to include('Main is not included in the list')
       end
     end
 
     context 'when contact_id not present' do
+      before { subject.contact_id = nil }
+
       it 'is not valid without an user_id' do
-        subject.contact_id = nil
-        expect { subject.valid? }.to raise_error("undefined method `phone_numbers' for nil:NilClass")
+        expect { subject.save! }.to raise_error(NoMethodError, "undefined method `phone_numbers' for nil:NilClass")
+      end
+    end
+  end
+
+  describe '#save' do
```

Comment:
It seems to be a test to ensure your callbacks `ensure_only_one_phone_number_is_main` and `set_main_if_phone_numbers_is_empty`, right? If yes, here you can split it into describe blocks, like so:

```rb
describe '#ensure_only_one_phone_number_is_main' do
  ...
end

describe '#set_main_if_phone_numbers_is_empty' do
  ...
end
```